src/paint.js

```diff
@@ -1,14 +1,14 @@
-import ThemeManager from './theme';
+import ThemeManager from "./theme";
 
-export default function($element, layout) {
+export default function ($element, layout) {
   let app = this.$scope.qlik.currApp(this);
   let isEditMode = this.$scope.options.interactionState === 2;
   this.$scope.updatedData(layout, isEditMode, true);
-  app.theme.getApplied().then(qTheme => {
+  return app.theme.getApplied().then(async (qTheme) => {
     this.$scope.theme = qTheme;
     ThemeManager.setAppTheme(this.$scope.theme);
 
-    this.$scope.updatedData(layout, isEditMode, true);
+    await this.$scope.updatedData(layout, isEditMode, true);
     return this.$scope.qlik.Promise.resolve();
```

Comment:
just return on line 11 and remove the last return?

Reply:
Oh that might work actually. Let me check